src/base/dof_map_constraints.C

```diff
@@ -2180,6 +2180,24 @@ void DofMap::enforce_constraints_on_residual (const NonlinearImplicitSystem & sy
   if (!solution)
     solution = system.solution.get();
 
+  NumericVector<Number> const * solution_local  = nullptr; // will be initialized below
+  std::unique_ptr<NumericVector<Number>> solution_built;
+  if (solution->type() == SERIAL || solution->type() == GHOSTED)
+      solution_local = solution;
+  else if (solution->type() == PARALLEL)
```

Comment:
Wait.  What's going on here @lindsayad ?  We definitely don't want to serialize solution vectors for constraint enforcement!  This will be a huge scaling problem.

Reply:
Well if it's a huge problem now, then it was also a huge problem before. See [here](https://github.com/libMesh/libmesh/pull/1986/files#diff-b3985fa832b099b796eaea1581eb1995L948)

Reply:
I see, @friedmud already knew the problem, but the code still got in!